module-dir-structure.md

```diff
@@ -17,21 +18,219 @@ my-new-module
+---
+```
+The `category` field should match one of the categories defined in the `config.json` file (`"categoryOrder"` field), which is located in the content's repo root directory; see, *e.g.*, [here](https://github.com/esciencecenter-digital-skills/NEBULA-content-template/blob/main/config.json). These categories are essentially sections that group together modules within specific thematic areas.
+
+The `id` field is used to determine the position of the current module within the specified `category`.
+
+The `title` field should be the title of the module; it will be shown on the main overview page.
+
+The `thumbnail` field should be the name of the thumbnail image, which should be placed in the `/modules/my-new-module/media` directory. This image shows up on the index page. You can add your own image to `/modules/my-new-module/media` and replace `thumbnail: "nlesc-dummy.png"` by `thumbnail: "my-module-thumbnail-image.png"`.
+
+The `visibility` field should be set to `visible` if you want to make this specific module visible in the final lesson rendering. Otherwise, set it to, *e.g.*, `not visible`.
+
+Finally, the `author`, `trl` (technical readiness level) and `abstract` fields do not influence the final module rendering, but they can be very useful for internal control and for developers.
 
 ## The `info.md` file
+This file defines the learning objectives of your module and is generally the first chapter to appear on the main module's overview page. An example `info.md` file is shown below:
+```markdown
+---
+title: Learning objectives
+type: info
+order: 0
+---
+
+- Appreciate the importance of testing software
+- Understand the various benefits of testing
+- Understand the types of tests and what info they convey
+- Get familiar with the idea of continuous integration and its importance
+```
+
+Here, the `title` corresponds to the title of your chapter as shown in the module's overview page.
+
+Within a module, the `type` field defines the chapter type and assumes the following options:
+- `info`: for chapters containing general information material.
+- `slides`: for chapters defining presentations; see example [here](#the-slides-pmd-file).
+- `reading`: for chapters containing focused reading material and on-line resources; see example [here](#exercises-and-online-resources-files).
+- `exercise`: for exercises; see example [here](#exercises-and-online-resources-files)
+
+Finally, the `order` field determines the sequence in which chapters appear on the module's overview page. Since the learning objectives are expected to come first, `order` should be set to `0` in the `info.md` file.
 
 ## The slides pmd file
 
-## Exercises and online resources files
+Similar to `info.md`, the slides file is also embedded as a chapter on the module's main page, using the `slides` chapter type, as described [above](#the-infomd-file).
+
+For now, they should be named with a different extension, `*.pmd`, to differentiate them from other chapter types.
```

Comment:
The slides pmd extension is not longer necessary since https://github.com/esciencecenter-digital-skills/NEBULA/pull/51

However the potentially confusing behavior that occurs if users forget to add frontmatter, or misformat it,  should be clearly documented here. Perhaps with a warning, e.g. : "Is your first slide not rendering correctly? Make sure that you have your front matter included and formatted like the following example."

Reply:
Thanks @JaroCamphuijsen for very useful comment. I have made the proper changes. I will proceed to your next review..

Reply:
Thanks.! I will also update the new slides `===` separator...doing right now!

Reply:
On the other PR. Sorry for my mess; should have done everything in just on PR.